pkg/api/v1alpha1/nutanixdatacenterconfig_test.go

```diff
@@ -0,0 +1,140 @@
+		expectedErr string
+		assertions  func(*testing.T, *NutanixDatacenterConfig)
+	}{
+		{
+			name:        "non-existent-file",
+			fileName:    "testdata/nutanix/non-existent-file.yaml",
+			expectedErr: "open testdata/nutanix/non-existent-file.yaml: no such file or directory",
+		},
+		{
+			name:        "invalid-file",
+			fileName:    "testdata/invalid_format.yaml",
+			expectedErr: "unable to parse testdata/invalid_format.yaml",
+		},
+		{
+			name:        "invalid-cluster-extraneous-field",
+			fileName:    "testdata/nutanix/invalid-cluster.yaml",
+			expectedErr: "unknown field \"idont\"",
+		},
+		{
+			name:        "valid-cluster",
+			fileName:    "testdata/nutanix/valid-cluster.yaml",
+			expectedErr: "",
+			assertions: func(t *testing.T, dcConf *NutanixDatacenterConfig) {
+				require.NoError(t, dcConf.Validate())
+				assert.Equal(t, expectedDCConf, dcConf)
+			},
+		},
+		{
+			name:        "valid-cluster-extra-delimiter",
+			fileName:    "testdata/nutanix/valid-cluster-extra-delimiter.yaml",
+			expectedErr: "",
+			assertions: func(t *testing.T, dcConf *NutanixDatacenterConfig) {
+				require.NoError(t, dcConf.Validate())
+			},
+		},
+		{
+			name:        "valid-cluster-setters-getters",
+			fileName:    "testdata/nutanix/valid-cluster.yaml",
+			expectedErr: "",
+			assertions: func(t *testing.T, dcConf *NutanixDatacenterConfig) {
+				assert.Equal(t, dcConf.ExpectedKind(), dcConf.Kind())
+
+				assert.False(t, dcConf.IsReconcilePaused())
+				dcConf.PauseReconcile()
+				assert.True(t, dcConf.IsReconcilePaused())
+				dcConf.ClearPauseAnnotation()
+				assert.False(t, dcConf.IsReconcilePaused())
+			},
+		},
+		{
+			name:        "valid-cluster-marshal",
+			fileName:    "testdata/nutanix/valid-cluster.yaml",
+			expectedErr: "",
+			assertions: func(t *testing.T, dcConf *NutanixDatacenterConfig) {
+				m := dcConf.Marshallable()
+				require.NotNil(t, m)
+				y, err := yaml.Marshal(m)
+				assert.NoError(t, err)
+				assert.NotNil(t, y)
+			},
+		},
+		{
+			name:        "valid-cluster-valid-trust-bundle",
+			fileName:    "testdata/nutanix/valid-cluster-valid-trustbundle.yaml",
+			expectedErr: "",
+			assertions: func(t *testing.T, dcConf *NutanixDatacenterConfig) {
+				assert.NoError(t, dcConf.Validate())
+			},
+		},
+		{
+			name:        "valid-cluster-invalid-trust-bundle",
+			fileName:    "testdata/nutanix/valid-cluster-invalid-trustbundle.yaml",
+			expectedErr: "",
+			assertions: func(t *testing.T, dcConf *NutanixDatacenterConfig) {
+				err := dcConf.Validate()
+				assert.Error(t, err)
+				assert.Contains(t, err.Error(), "could not find a PEM block in the certificate")
+			},
+		},
+		{
+			name:        "invalid-kind",
+			fileName:    "testdata/nutanix/invalid-kind.yaml",
+			expectedErr: "does not contain kind NutanixDatacenterConfig",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			conf, err := GetNutanixDatacenterConfig(test.fileName)
+			if test.expectedErr == "" {
```

Comment:
Love the use of table tests. 
But could we avoid `if else` clauses and break these into different unit tests? Table tests are best used for varying input for a single scenario.

Reply:
Not sure I subscribe to that ideology but I'll break it up for the purpose of moving the PR forward.